Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.mm

```diff
@@ -0,0 +1,321 @@
+            m_data.allowedProtocolTypes = static_cast<NSURLRequestAllowedProtocolTypes>(val);
+    }
+
+    SET_NSURLREQUESTDATA(boundInterfaceIdentifier, NSString, CoreIPCString);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsExpensiveNetworkAccess, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsConstrainedNetworkAccess, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsUCA, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(assumesHTTP3Capable, NSNumber, bool);
+
+    id attribution = dict[@"attribution"];
+    if (attribution != nil && [attribution isKindOfClass:[NSNumber class]]) {
+        auto val = [allowedProtocolTypes unsignedCharValue];
+        if (isValidEnum<NSURLRequestAttribution>(val))
+            m_data.attribution = static_cast<NSURLRequestAttribution>(val);
+    }
+
+    SET_NSURLREQUESTDATA_PRIMITIVE(knownTracker, NSNumber, bool);
+    SET_NSURLREQUESTDATA(trackerContext, NSString, CoreIPCString);
+    SET_NSURLREQUESTDATA_PRIMITIVE(privacyProxyFailClosed, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(privacyProxyFailClosedForUnreachableNonMainHosts, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(privacyProxyFailClosedForUnreachableHosts, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(requiresDNSSECValidation, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsPersistentDNS, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(prohibitPrivacyProxy, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowPrivateAccessTokensForThirdParty, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(useEnhancedPrivacyMode, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(blockTrackers, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(failInsecureLoadWithHTTPSDNSRecord, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(isWebSearchContent, NSNumber, bool);
+
+    NSArray *contentDispositionEncodingFallbackArray = dict[@"contentDispositionEncodingFallbackArray"];
+    if (contentDispositionEncodingFallbackArray != nil && [contentDispositionEncodingFallbackArray isKindOfClass:[NSArray class]]) {
+        Vector<CoreIPCNumber> vector;
+        vector.reserveInitialCapacity(contentDispositionEncodingFallbackArray.count);
+        for (id element in contentDispositionEncodingFallbackArray) {
+            if ([element isKindOfClass:[NSNumber class]]) {
+                CoreIPCNumber tooAdd(element);
+                vector.append(WTFMove(tooAdd));
+            }
+        }
+        vector.shrinkToFit();
+        m_data.contentDispositionEncodingFallbackArray = WTFMove(vector);
+    }
+}
+
+CoreIPCNSURLRequest::CoreIPCNSURLRequest(CoreIPCNSURLRequestData&& data)
+    : m_data(WTFMove(data)) { }
+
+CoreIPCNSURLRequest::CoreIPCNSURLRequest(const RetainPtr<NSURLRequest>& request)
+    : CoreIPCNSURLRequest(request.get()) { }
+
+
+RetainPtr<id> CoreIPCNSURLRequest::toID() const
+{
+    RetainPtr<NSMutableDictionary> dict = adoptNS([[NSMutableDictionary alloc] initWithCapacity:19]);
```

Comment:
nit: I think the deduction guides on RetainPtr make it so you don't need <NSMutableDictionary> a few times in this function.